### PR TITLE
DCS-519 Adding new endpoint to retrieve offenders who entered prison in a time frame

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+@AllArgsConstructor
+@Entity
+@Table(name = "OFFENDER_EXTERNAL_MOVEMENTS")
+@IdClass(ExternalMovement.PK.class)
+public class ExternalMovement extends AuditableEntity {
+
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class PK implements Serializable {
+        @Column(name = "OFFENDER_BOOK_ID", updatable = false, insertable = false)
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java:42]: You can map to complex types for keys as well if you want...

File: src/main/java/net/syscon/prison/repository/jpa/model/Offender.java [hunk 2]
```diff
@@ -45,9 +45,19 @@
...
+    private String firstName;
+
+    @Column(name = "MIDDLE_NAME", nullable = true)
+    private String middleName;
+
     @Column(name = "LAST_NAME", nullable = false)
     private String lastName;
 
+    @Column(name = "BIRTH_DATE", nullable = true)
+    @CreatedDate
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/prison/repository/jpa/model/Offender.java:58]: Did you mean for this `@CreatedDate` annotation here?

File: src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import java.io.Serializable;
+import java.time.LocalDateTime;
+
+import static javax.persistence.EnumType.STRING;
+import static net.syscon.prison.repository.jpa.model.City.CITY;
+import static net.syscon.prison.repository.jpa.model.MovementReason.REASON;
+import static net.syscon.prison.repository.jpa.model.MovementType.TYPE;
+import static org.hibernate.annotations.NotFoundAction.IGNORE;
+
+@Data
```
Comment (matt-crowson-moj) [inline src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java:30]: Could also add `@EqualsAndHashCode(callSuper = false)`

File: src/main/java/net/syscon/prison/repository/jpa/model/Offender.java (hunk 2 above)
Comment (andrewrlee) [reply src/main/java/net/syscon/prison/repository/jpa/model/Offender.java]: removed

File: src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java (hunk 3 above)
Comment (andrewrlee) [reply src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java]: what would that do?

File: src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java (hunk 1 above)
Comment (andrewrlee) [reply src/main/java/net/syscon/prison/repository/jpa/model/ExternalMovement.java]: as discussed, that doesn't seem to work in this case.

**Pull request merged**